cpp/test/adapter-eth-eos.test.js

```diff
@@ -0,0 +1,281 @@
+const { expect } = require('chai')
+const { Blockchain, expectToThrow, mintTokens } = require('@eosnetwork/vert')
+const { deploy } = require('./utils/deploy')
+const { Asset, Bytes, PublicKey } = require('@wharfkit/antelope')
+const R = require('ramda')
+const {
+  active,
+  precision,
+  getAccountCodeRaw,
+  getSymbolCodeRaw,
+  getSingletonInstance,
+  logExecutionTraces,
+  prettyTrace,
+} = require('./utils/eos-ext')
+const { getEventBytes } = require('./utils/get-event-bytes')
+const { substract, no0x } = require('./utils/wharfkit-ext')
+const { getAccountsBalances } = require('./utils/get-token-balance')
+const { getMetadataSample } = require('./utils/get-metadata-sample')
+const { getOperationSample } = require('./utils/get-operation-sample')
+const errors = require('./utils/errors')
+
+const ethers = require('ethers')
+
+const getSwapMemo = (sender, destinationChainId, recipient, data) =>
+  `${sender},${destinationChainId},${recipient},${R.isEmpty(data) ? '0' : '1'}`
+
+const attestation = 'deadbeef'
+
+const hexStringToBytes = (hex) => {
+    // Ensure the input string is valid
+    if (hex.length % 2 !== 0) {
+        throw new Error("Hex string must have an even length.");
+    }
+
+    const bytes = [];
+    for (let i = 0; i < hex.length; i += 2) {
+        // Parse two hex characters at a time and convert to a byte (0-255)
+        bytes.push(parseInt(hex.substr(i, 2), 16));
+    }
+    return bytes;
+}
+
+describe('Adapter EVM -> EOS testing', () => {
+  const symbol = 'TKN'
+  const minFee = `0.001800 X${symbol}`
+  const precision4 = precision(4)
+  const precision6 = precision(6)
+  const precision18 = precision(18)
+  const maxSupply = '500000000.000000'
+  const tokenMaxSupply = '500000000.000000000000000000'
+  const userInitialBalance = `1000.0000 ${symbol}`
+  const tokenBytes = no0x(
+    ethers.zeroPadValue(
+      ethers.toBeHex(getSymbolCodeRaw(`0.0000 ${symbol}`).toString()),
+      32,
+    ),
+  )
+
+  const TABLE_STORAGE = 'storage'
+  const FEE_BASIS_POINTS = 1750
+  const FEE_BASIS_POINTS_DIVISOR = 1000000
+
+  const token = {
+    symbol: symbol,
+    account: `${symbol.toLowerCase()}.token`,
+    maxSupply: `${maxSupply} ${symbol}`,
+    bytes: '0000000000000000000000003ca5269b5c54d4c807ca0df7eeb2cb7a5327e77d', //tokenBytes,
```

Comment:
Easier for the reader
```suggestion
    bytes: '0000000000000000000000003ca5269b5c54d4c807ca0df7eeb2cb7a5327e77d', // eth address bytes representation
```